### PR TITLE
feat: add specific directories to .gitignore when calling snforge init

### LINKED ISSUE #2298: Add specific directories to `.gitignore` when calling `snforge init`
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

When calling `snforge new <project>` there is a `.gitignore` file created, that states what files should be omitted by the vcs. It is, however, not complete as it omits a few files that snforge is able to create during its runs, which may result in accidental inclusion of this files by git. 

We should modify the `snforge new` code, so it includes these files in the `.gitignore` file:
- `snfoundry_trace`
- `.snfoundry_cache`
- `coverage`
- `profile`

This should be done in this function: https://github.com/foundry-rs/starknet-foundry/blob/8ce451e58b1305f7e6769451a18412970157dc1a/crates/forge/src/new.rs#L145

Definition of done:
- Running `snforge new <project>` creates a `.gitignore` file that includes paths specified in this issue
- There are e2e tests to make sure the changes are working

### PR DESCRIPTION
**Pull request by Akshola00**
# Update `.gitignore` in `snforge init` to Include Additional Files

## Description

This PR modifies the `snforge init` functionality to ensure that the `.gitignore` file generated by `snforge` includes additional paths for files that may be created during its execution. These files are often specific to `snforge` and can otherwise be mistakenly committed to version control.

Closes #2298

## Changes

The following updates were made to `snforge init`:

- Modified the code to append the following paths to the `.gitignore`:
  - `snfoundry_trace`
  - `.snfoundry_cache`
  - `.snfoundry_versioned_programs`
  - `coverage`
  -  Added end-to-end (e2e) tests to validate that running `snforge init <project>` now generates a `.gitignore` file containing all required paths.

## Checklist

- [x] Modify `snforge init` to include additional files in `.gitignore`.
- [x] Add e2e tests to confirm changes are effective.
- [x] Verify that `snforge init <project>` correctly generates a `.gitignore` with all specified paths.

### CODE REVIEW

**Comment by Akshola00** [comment]
@THenry14 pls review

**Comment by Akshola00** [comment]
Any updates on the review: @THenry14 @cptartur @ddoktorski @kkawula @franciszekjob

**Comment by THenry14** [comment]
Hi @Akshola00 , thanks for the PR. Please fix the issues with formating and linter reported by CI. Thanks.

File: crates/forge/tests/e2e/running.rs [hunk 1]
```diff
@@ -806,6 +808,26 @@ fn validate_init(temp: &TempDir, validate_snforge_std: bool) {
...
+    ];
+
+    for entry in expected_entries {
+        assert!(
+            gitignore_content.contains(entry),
+            "Expected .gitignore to contain '{}', but it didn't",
+            entry
+        );
+    }
+}
```
Comment (THenry14) [inline crates/forge/tests/e2e/running.rs:829]: I think you should be able to get rid of this `for` loop if you used `indoc!` to hold expected entries. 
hint: you can try and compare the whole file

**Comment by THenry14** [comment]
> @THenry14 formatted code

Great, thanks. You forgot about linter though https://github.com/foundry-rs/starknet-foundry/actions/runs/11551267348/job/32148063530?pr=2611

You can ingore the `CI/Test building docs run`

**Review by THenry14 — CHANGES_REQUESTED** [review]
Hi @Akh

**Comment by Akshola00** [comment]
making the necessary changes now

File: crates/forge/tests/e2e/running.rs [hunk 2]
```diff
@@ -806,6 +808,26 @@ fn validate_init(temp: &TempDir, validate_snforge_std: bool) {
     assert_stdout_contains(output, expected);
 }
 
+fn validate_gitignore(temp: &TempDir) {
+    let gitignore_path = temp.join("test_name/.gitignore");
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply crates/forge/tests/e2e/running.rs]: reloved

File: crates/forge/tests/e2e/running.rs (hunk 1 above)
Comment (Akshola00) [reply crates/forge/tests/e2e/running.rs]: resollved

**Comment by Akshola00** [comment]
Made the necessary changes, @THenry14. The CI issue has been fixed because I'm no longer using that syntax.

File: crates/forge/tests/e2e/running.rs [hunk 3]
```diff
@@ -806,6 +808,24 @@ fn validate_init(temp: &TempDir, validate_snforge_std: bool) {
     assert_stdout_contains(output, expected);
 }
 
+fn validate_gitignore(temp: &TempDir, path: &str) {
```
Comment (THenry14) [inline crates/forge/tests/e2e/running.rs:811]: you don't need to pass the path as separate parameter, you can just pass the path as one

Comment (Akshola00) [reply crates/forge/tests/e2e/running.rs]: done

**Comment by Akshola00** [comment]
@THenry14 any other changes

**Review by THenry14 — CHANGES_REQUESTED** [review]
Please resolve issues reported by CI

**Review by cptartur — COMMENTED** [review]
Hey @Akshola00, please update this PR so problems with unsuccessful checks are resolved.

**Comment by Akshola00** [comment]
> Hey @Akshola00, please update this PR so problems with unsuccessful checks are resolved.

Apologies for the long silence
Working on it now and will resolve changes between today and early tomorrow

File: crates/forge/tests/e2e/running.rs [hunk 4]
```diff
@@ -806,6 +808,24 @@ fn validate_init(temp: &TempDir, validate_snforge_std: bool) {
...
 
+fn validate_gitignore(path: &Path) {
+    let gitignore_path = path.join(".gitignore");
+    let gitignore_content = fs::read_to_string(gitignore_path).unwrap();
+
+    let expected_gitignore = indoc! {"
+        target
+        .snfoundry_cache/
+        snfoundry_trace/
+        snfoundry_versioned_programs/
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply crates/forge/tests/e2e/running.rs:819]: updated the syntax
curretnly fixing the CI issues

**Comment by Akshola00** [comment]
any hints on how to resolve the CI

**Comment by cptartur** [comment]
> im having trouble resolving the CI lint issue any hints on how to go about it

You can run `cargo lint` and address any issues reported by the compiler. Make sure you are on latest version of rust.

Same goes for failing forge tests `cargo test -p forge`.

**Comment by Akshola00** [comment]
@THenry14 pls review

**Comment by Akshola00** [comment]
@THenry14 @cptartur any updates on the review

**Comment by THenry14** [comment]
> @THenry14 @cptartur any updates on the review

Hi @Akshola00 , your tests sadly still are not passing :(

Please make sure they work **before** you push changes. It will save you (and us) a lot of time. thanks.

File: crates/forge/tests/e2e/running.rs [hunk 5]
```diff
@@ -726,14 +726,27 @@ pub fn append_to_path_var(path: &Path) -> OsString {
...
     } else {
         ""
     };
 
+    let expected_gitignore = formatdoc!(
+        r#"
+        .snfoundry_cache/
+        snfoundry_trace/ 
+        .snfoundry_versioned_programs/
+        coverage/∅
```
Comment (cptartur) [inline crates/forge/tests/e2e/running.rs:744]: I don't think this is valid

**Review by cptartur — CHANGES_REQUESTED** [review]

**Comment by cptartur** [comment]
Closing due to not addressing requests to fix tests in multiple reviews.

**Comment by Akshola00** [comment]
image

never added this to my code but it keeps showing up

**Pull request closed without merging**